org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/SynapseLanguageService.java

```diff
@@ -669,6 +672,27 @@ public CompletableFuture<String> downloadDriverForConnector(DriverDownloadReques
                 request.getConnectionType()));
     }
 
+    @Override
+    public CompletableFuture<DeployPluginDetails> updateMavenDeployPlugin(DeployPluginDetails pluginDetails) {
+
+        DeployPluginDetails response = PomParser.addCarDeployPluginToPom(new File(projectUri + File.separator + Constants.POM_FILE), pluginDetails);
+        return CompletableFuture.supplyAsync(() -> response);
```

Comment:
```suggestion
        return CompletableFuture.supplyAsync(() -> PomParser.addCarDeployPluginToPom(new File(projectUri + File.separator + Constants.POM_FILE), pluginDetails));
```

check other methods too.